app/controllers/admin/collections_controller.rb

```diff
@@ -326,7 +326,11 @@ def build_default_lending_period(collection)
       collection.errors.add(:lending_period, "hours needs to be a positive integer.")
     end
 
-    flash[:notice] = collection.errors.full_messages.join if collection.errors.present?
+    if lending_period.zero?
+      collection.errors.add(:lending_period, "cannot be set to 0 days 0 hours.")
+    end
+
+    flash[:notice] = collection.errors.full_messages.join if collection.errors.present? && Settings.controlled_digital_lending.enable
     lending_period.to_i
   end
```

Comment:
I can't figure out how to highlight the rest of the build_default_lending_period method for comment, but there is a similar method in the [access_control_step](https://github.com/avalonmediasystem/avalon/blob/fc7bba2bd8efd2436e872fa05dd260147dbbceba/app/models/access_control_step.rb#L129) that is a lot cleaner. The error handling is done when the method is called around line 102. 

@cjcolvar, is there any reason you ran into or can see to not change this to match the implementation for building and setting the lending period in the access_control_step?

Reply:
I went ahead and did this refactoring as part of the work on #4809. Everything seems good so far, but still let me know if you have any thoughts.